addons/mail/static/src/models/messaging/messaging.js

```diff
@@ -15,8 +17,8 @@ function factory(dependencies) {
          * @override
          */
         _willDelete() {
-            if (this.env.services['bus_service']) {
-                this.env.services['bus_service'].off('window_focus', null, this._handleGlobalWindowFocus);
+            if (owl.Component.env.services['bus_service']) {
```

Comment:
It's alright the way it is now, but I think we could improve code maintainability of handling envs in discuss code by introducing single points of entry to get them. For instance:

- `this.env` in models/components becomes `this.messaging.env`.
- `owl.Component/env` in models/components becomes `this.messaging.legacyEnv`.

I think doing so makes all our code much more cohesive, as every entry point comes from messaging, even access to code outside of models.